packages/headless/src/database/mongodb/search/index.ts

```diff
@@ -0,0 +1,10 @@
+import { Injectable } from "@nestjs/common";
+import { Product } from "src/entity/product";
+import { ProductModel } from "../product/product.model";
+
+@Injectable()
+export class ProductSrarchDatabase {
+    async findAllProducts(): Promise<Product[]> {
+        return await ProductModel.find({}).lean();
+    }
```

Comment:
for this type of read operation please use pagination. otherwise when your product list will grow up, the more RAM might be needed to load the whole data.

Reply:
Fixed

Reply:
delete this function

Reply:
done